lib/modules.nix

```diff
@@ -442,7 +443,8 @@ rec {
         }
       */
       byName = attr: f: modules:
-        foldl' (acc: module:
+        zipAttrsWith (n: concatLists)
+          (map (module:
               if !(builtins.isAttrs module.${attr}) then
                 throw ''
                   You're trying to declare a value of type `${builtins.typeOf module.${attr}}'
```

Comment:
This function is used on both `options` and `config`, yet the message is the same. This can't be right.
Perhaps the function should be duplicated, specializing for `config` and `options` respectively.

Reply:
turning zipAttrsWith into a primop is massively more efficient and benefits other users too (eg recursiveUpdate). we've made a prototype implementation of `__zipAttrsWith` and have seen another sizable decrease in resource usage on system builds :)